service/src/main/java/bio/terra/catalog/service/DatasetService.java

```diff
@@ -131,4 +140,90 @@ public DatasetId createDataset(
     ensureActionPermission(user, dataset, SamAction.CREATE_METADATA);
     return datasetDao.create(dataset).id();
   }
+
+  public DatasetPreviewMetadataResponse getDatasetPreviewMetadata(
+      AuthenticatedUserRequest user, DatasetId datasetId) {
+    var dataset = datasetDao.retrieve(datasetId);
+    ensureActionPermission(user, dataset, SamAction.READ_ANY_METADATA);
+    // Right now, this code makes an assumption that all datasets are snapshots.
+    // Eventually this will not be true and this should change.
+    var snapshotMetadata = datarepoService.getPreviewMetadata(user, dataset.storageSourceId());
+    return new DatasetPreviewMetadataResponse()
+        .id(snapshotMetadata.getId())
+        .dataProject(snapshotMetadata.getDataProject())
+        .tables(convertDatarepoTablesToCatalogTables(snapshotMetadata.getTables()));
+  }
+
+  private List<TableModel> convertDatarepoTablesToCatalogTables(
+      List<bio.terra.datarepo.model.TableModel> datarepoTables) {
+    return Optional.ofNullable(datarepoTables)
```

Comment:
In general I don't think `Optional.ofNullable()` should be used as a replacement for a conditional. If you want to write this in a functional-ish way, you can use `Objects.requireNonNullElse()`. Personally I would use `!= null`.

Although, is it possible for `datrepoTables` to be `null` in practice? Ask in jade-core to be sure, but I don't think it is.